src/Orchard.Web/Modules/Orchard.Taxonomies/Drivers/TaxonomyFieldDriver.cs

```diff
@@ -70,9 +70,11 @@ public class TaxonomyFieldDriver : ContentFieldDriver<TaxonomyField> {
                     DisplayName = field.DisplayName,
                     Name = field.Name,
                     Terms = terms,
+                    SelectedTerms = appliedTerms.Select(t => t.Value),
                     Settings = settings,
-                    SingleTermId = terms.Where(t => t.IsChecked).Select(t => t.Id).FirstOrDefault(),
-                    TaxonomyId = taxonomy != null ? taxonomy.Id : 0
+                    SingleTermId = appliedTerms.Select(t => t.Key).FirstOrDefault(),
+                    TaxonomyId = taxonomy != null ? taxonomy.Id : 0,
+                    HasTerms = taxonomy != null && _taxonomyService.GetTermsCount(taxonomy.Id) > 0

```

Comment:
`terms` will only be populated when not using autocomplete, otherwise we don't bother loading all the terms